docs/blog-1/blog-1.md

```diff
@@ -0,0 +1,253 @@
+While the PR was under review, you started work on another topic branch `block-cancel-order`.
+
+Unfortunately, the changes that you intended to introduce on `block-cancel-order` depended on what had already been done on `adjust-reads-prec`...
+So you forked a new branch (`block-cancel-order`) off `adjust-reads-prec` and, when the change was ready, you made another PR, this time for `block-cancel-order` to `adjust-reads-prec`.
+
+In the meantime, reviewers posted their comments on the first PR.
+You applied their remarks as a `1st round of fixes` commit on the `adjust-reads-prec` branch.
+
+Let's see how the state of the repo looks in a graphical tool like `gitk`:
+
+![gitk](gitk-0.png)
+
+In most cases it's rather unusual to have more than 2 branches stacked on top of each other, but for the sake of completeness let's now make this chain longer.
+You developed a couple of new refactors and bug fixes - say, on branches `change-table` and `drop-location-type`.
+As each of them was dependent on changes already waiting in the review queue, you began stacking another branches on top of each other.
+You ended up having 4 branches in the chain, each dependent on the previous one: `adjust-reads-prec`, `block-cancel-order`, `change-table` and `drop-location-type`.
+
+Now for the sake of providing a full-fledged example, let's also add a couple of other branches (not related to the 4-branch chain) to our scenario.
+Let's assume you also independently developed a feature `edit-margin-not-allowed` and later derived a branch `full-load-gatling` from that point.
+Also, you created a branch `grep-errors-script` that (fortunately!) nothing depended on, and a `hotfix/remove-trigger` branch, but on top of `master`, not `develop`.
+
+If you'd like to quickly recreate this repository state on your local machine, you can use [this handy script (link)](https://raw.githubusercontent.com/VirtusLab/git-machete/copy-docs/docs/blog-1/sandbox-setup.sh),
+which will set it up in `machete-sandbox` directory in your home folder.
+
+
+Now the problem: how to quickly check which of the branches are in sync with their parent (aka upstream) branches?
+Note that for some of them (like `adjust-reads-prec` or `hotfix/remove-trigger`) this parent would simply be `develop` or `master`, but for others (like `change-table`) this would be another topic branch.
+Also, how to easily rebase each branch on the top of its parent, especially when dependencies between branches have changed from the description above, or some branches have been deleted, and so on?
+
+
+## Defining a structure for the branches (`edit` subcommand)
+
+
+Okey, let's get our hands dirty... first install `git-machete` with the sequence of commands which fits your Operating System that you can also find in the README of
+[the git machete repo (github.com/VirtusLab/git-machete)](https://github.com/VirtusLab/git-machete).
```

Comment:
Pls add `#`-fragment so that the link points directly to the installation manual